tools/Jenkins/dependency_spotter.py

```diff
@@ -0,0 +1,143 @@
+import argparse
```

Comment:
Needs to have the copyright header (see `tools/Cppcheck/generate_cppcheck_suppressions_list.py` for an example / something to copy)
Would also be good to have a doc string to explain what this script does and maybe some example usage.